### PR TITLE
Bluetooth: conn: Don't deadlock on `BT_RECV_WORKQ_SYS`

### PR DESCRIPTION
**Pull request by jori-nordic**
My recent change overlooked this configuration, which is not tested in CI currently.

If someone had enabled it, they would've just gotten an unfixable assert.

The correct test would be checking if the current thread is the system wq, but I think the compiler can't know that and can't remove the "dead" code.

### CODE REVIEW

**Review by Thalley — DISMISSED** [review]

**Review by alwa-nordic — DISMISSED** [review]

**Review by Thalley — DISMISSED** [review]

File: subsys/bluetooth/host/conn.c [hunk 1]
```diff
@@ -377,15 +377,23 @@ static void bt_acl_recv(struct bt_conn *conn, struct net_buf *buf,
...
 #if defined(CONFIG_BT_CONN_TX)
-	struct k_work_sync sync;
+	LOG_DBG("conn %p", conn);
 
-	/* API docs mention undefined behavior if syncing on work item from wq
-	 * execution context.
-	 */
-	__ASSERT_NO_MSG(k_current_get() != &k_sys_work_q.thread);
+	if (IS_ENABLED(CONFIG_BT_RECV_WORKQ_SYS)) {
+		__ASSERT_NO_MSG(k_current_get() == &k_sys_work_q.thread);
```
Comment (jori-nordic) [inline subsys/bluetooth/host/conn.c:383]: maybe I can just put this in `tx_notify`. I'll do that if I have to push again :)

**Comment by jori-nordic** [comment]
I will post another PR that enables that config for the l2cap stress test so we get some coverage.

**Comment by jori-nordic** [comment]
> If someone had enabled it, they would've just gotten an unfixable assert

yupp
image

**Review by Thalley — DISMISSED** [review]

File: subsys/bluetooth/host/conn.c [hunk 2]
```diff
@@ -377,15 +377,23 @@ static void bt_acl_recv(struct bt_conn *conn, struct net_buf *buf,
...
-	k_work_submit(&conn->tx_complete_work);
-	k_work_flush(&conn->tx_complete_work, &sync);
+		tx_notify(conn);
+	} else {
+		struct k_work_sync sync;
+
+		/* API docs mention undefined behavior if syncing on work item
+		 * from wq execution context.
+		 */
+		__ASSERT_NO_MSG(k_current_get() != &k_sys_work_q.thread);
```
Comment (alwa-nordic) [inline subsys/bluetooth/host/conn.c:392]: Consider using the getter `k_work_queue_thread_get`

Comment (jori-nordic) [reply subsys/bluetooth/host/conn.c:392]: no one uses it but ok

File: subsys/bluetooth/host/conn.c (hunk 1 above)
Comment (jori-nordic) [reply subsys/bluetooth/host/conn.c:383]: done

**Review by alwa-nordic — DISMISSED** [review]

**Comment by jori-nordic** [comment]
image

C in a nutshell:

image

**Review by alwa-nordic — APPROVED** [review]

**Review by Thalley — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**